src/wayland/selection/wlr_data_control/mod.rs

```diff
@@ -0,0 +1,229 @@
+//!
+//! ## Initialization
+//!
+//! To initialize this implementation, create [`DataControlState`], store it in your `State`
+//! struct, and implement the required trait, as shown in the example:
+//!
+//! ```
+//! # extern crate wayland_server;
+//! # #[macro_use] extern crate smithay;
+//! use smithay::wayland::selection::SelectionHandler;
+//! use smithay::wayland::selection::wlr_data_control::{DataControlState, DataControlHandler};
+//! # use smithay::input::{Seat, SeatHandler, SeatState, pointer::CursorImageStatus};
+//! # use smithay::reexports::wayland_server::protocol::wl_surface::WlSurface;
+//!
+//! # struct State { data_control_state: DataControlState }
+//! # let mut display = wayland_server::Display::<State>::new().unwrap();
+//! // Create the data_control state
+//! let data_control_state = DataControlState::new::<State>(
+//!     &display.handle(), None
+//! );
+//!
+//! // insert the DataControlState into your state
+//! // ..
+//!
+//! // implement the necessary traits
+//! # impl SeatHandler for State {
+//! #     type KeyboardFocus = WlSurface;
+//! #     type PointerFocus = WlSurface;
+//! #     fn seat_state(&mut self) -> &mut SeatState<Self> { unimplemented!() }
+//! #     fn focus_changed(&mut self, seat: &Seat<Self>, focused: Option<&WlSurface>) { unimplemented!() }
+//! #     fn cursor_image(&mut self, seat: &Seat<Self>, image: CursorImageStatus) { unimplemented!() }
+//! # }
+//! impl SelectionHandler for State {
+//!     type SelectionUserData = ();
+//! }
+//! impl DataControlHandler for State {
+//!     fn data_control_state(&self) -> &DataControlState { &self.data_control_state }
+//!     // ... override default implementations here to customize handling ...
+//! }
+//! delegate_data_control!(State);
+//!
+//! // You're now ready to go!
+//! ```
+//!
+//! Be aware that data control clients rely on other selection providers to be implemneted, like
+//! wl_data_device or zwp_primary_selection.
+
+use wayland_protocols_wlr::data_control::v1::server::zwlr_data_control_manager_v1::ZwlrDataControlManagerV1;
+use wayland_server::backend::GlobalId;
+use wayland_server::{DisplayHandle, GlobalDispatch};
+
+mod device;
+mod source;
+
+pub use device::DataControlDeviceUserData;
+pub use source::DataControlSourceUserData;
+
+use super::primary_selection::PrimarySelectionState;
+use super::SelectionHandler;
+
+/// Access the data control state.
+pub trait DataControlHandler: Sized + SelectionHandler {
+    /// [DataControlStat] getter.
+    fn data_control_state(&self) -> &DataControlState;
+}
+
+/// State of the data control.
+#[derive(Debug)]
+pub struct DataControlState {
+    manager_global: GlobalId,
+}
+
+impl DataControlState {
+    /// Register new [ZwlrDataControlManagerV1] global.
+    ///
+    /// Passing `primary_selection` will enable support for primary selection as well.
+    pub fn new<D>(display: &DisplayHandle, primary_selection: Option<&PrimarySelectionState>) -> Self
```

Comment:
Can we please add a variant here, that allows setting a `filter` for which clients can bind to this global?

Reply:
Done.